### ISSUE TITLE
How to subscribe for realtime logs ?

### ISSUE THREAD

**Issue by imfaisii** [rabbygit/node_zklib#1]
Hello, I hope you are doing well. I am very new to nodejs . I am using your library , my device is connected . I can see the users and attendances but how am I going to get realtime logs. I am inside example and typed node test.js and it says Previous real time event removed!!!. and is on infinite wait. How will it trigger that realtime call ?

**Comment by rabbygit** [comment]
Isn't it triggered when a verified fingerprint is placed?

**Comment by imfaisii** [comment]
> Isn't it triggered when a verified fingerprint is placed?

Nope , always waiting . do i have to goto example folder and type node test ? that s is ? and on console it will show the log when the finger is placed ? after it says Previous real time event removed!!! ?

**Comment by rabbygit** [comment]
Yes, after displaying "Previous real time event removed!!!", it should print in the console if a verified finger is placed.

**Comment by imfaisii** [comment]
> Yes, after displaying "Previous real time event removed!!!", it should print in the console if a verified finger is placed.

oh, it is not working in my case , actually I am a laravel dev , i tried php library of it but that was not this much reliable. I will try to test it some more time , and 1 thing about deployment , I wont be able to deploy it on shared hosting right ? it will give port error , I will need a VPS for it ?

**Comment by rabbygit** [comment]
I think deployment in shared hosting won't cause any problem. Give it a try!

**Comment by rabbygit** [comment]
By the way, DId you set any password on your device? If yes then you need to pass the password during instance creation.

**Comment by imfaisii** [comment]
No it is not password protected

**Comment by rabbygit** [comment]
Today, I run into the same problem with one of the devices. After some debugging, I find that I did not register users first. Did you register users on your device? To check registered users run the following code:

```sh
const users = await zkInstance.getUsers()
console.log(users)
```

**Comment by imfaisii** [comment]
Yes I am doing this step , i can see users and logs but realtime logs isn't working

**Comment by daoues1** [comment]
I had faced the same problem, i can't get an acces to realtime logs , did you find any solution ? Thanks.

**Comment by Muddasir1** [comment]
Real-time attendance working fine in my case it working perfect 

Greate work
